Dynatrace/dynatrace-operator
# Feature: Adds automatic kubernetes api monitoring to dynakube helm chart

# Description
This feature request adds the `operator.dynatrace.com/feature-automatic-kubernetes-api-monitoring` annotation to the DynaKube helm chart when `activeGate.autoApiMonitoring` is true and `kubernetes-monitoring` is in activeGate.capabilities in values.yaml.

This enables the option to connect the Dynatrace Operator automatically to your Dynatrace environment, as defined at  https://www.dynatrace.com/support/help/setup-and-configuration/setup-on-container-platforms/kubernetes/connect-your-k8s-clusters#auto

https://github.com/Dynatrace/dynatrace-operator/issues/649

## How can this be tested?
In `values.yaml` set `autoApiMonitoring` under `activeGate` to `true` (false by default).
Then add `kubernetes-monitoring` to `capabilities` list under `activeGate` in `values.yaml` (Required).

Once changes are deployed check back in your Dynatrace environment in `Kubernetes` you should see your cluster.

**_Note: ensure you have done the prerequisites as per the documents https://www.dynatrace.com/support/help/setup-and-configuration/setup-on-container-platforms/kubernetes/set-up-k8s-monitoring#prerequisites_**

## Checklist
